vminfo_parser/vminfo_parser.py

```diff
@@ -480,13 +477,13 @@ def visualize_supported_os_distribution(
             plt.show(block=True)
             plt.close()
 
-    def visualize_os_version_distribution(self: t.Self, os_name: str) -> None:
-        counts = self.cli_output.generate_os_version_distribution(self.analyzer, os_name)
+    def visualize_os_version_distribution(self: t.Self, os_name: str, dataFrame: pd.DataFrame, config: Config) -> None:
+        counts = self.cli_output.generate_os_version_distribution(dataFrame, os_name, config)
 
         if not counts.empty:
             ax = counts.plot(kind="barh", rot=45)
 
-            if self.analyzer.config.generate_graphs:
+            if config.generate_graphs:
```

Comment:
same comment as above,  we should check `config.generate_graphs` before call calling visualizer, rather than passing it to visualizer to check.

Reply:
This is a good catch. I was moving them out and I missed these 2. Thanks, will fix